trillian/integration/hammer.go

```diff
@@ -197,25 +194,9 @@ type hammerState struct {
 }
 
 func newHammerState(cfg *HammerConfig) (*hammerState, error) {
-	opts := jsonclient.Options{}
-	if cfg.LogCfg.PubKeyPEMFile != "" {
-		pubkey, err := ioutil.ReadFile(cfg.LogCfg.PubKeyPEMFile)
-		if err != nil {
-			return nil, fmt.Errorf("failed to get public key contents: %v", err)
-		}
-		opts.PublicKey = string(pubkey)
-	}
-	var pool ClientPool
-	for _, s := range strings.Split(cfg.Servers, ",") {
-		c, err := client.New("http://"+s+"/"+cfg.LogCfg.Prefix, nil, opts)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create LogClient instance: %v", err)
-		}
-		pool = append(pool, c)
-	}
 	state := hammerState{
 		cfg:   cfg,
-		pool:  pool,
+		pool:  cfg.ClientPool,
```

Comment:
Seems superfluous to pull out one field from `cfg` for special treatment. OTOH, doing (say) `s.cfg.ClientPool.Next().AddChain(...)` everywhere is a bit ugly.  

Maybe go for a middle ground by adding a helper like:
```go
func (s *HammerState) client() *client.LogClient {
  return s.cfg.ClientPool.Next()
}
```
and then everything gets shorter (`s.client().AddChain(...)`)

Up to you.